### PR TITLE
Create exercise rail-fence-cipher

### PR DESCRIPTION
**Pull request by PaulT89**
## Summary

Generated practice exercise Rail Fence Cipher.

Difficulty = 4.  Hovers around here for other tracks.  Seems good enough.

## Checklist
- [x] If docs where changed run `./bin/configlet generate` to ensure all documents are properly generated.
- [x] CI is green

### CODE REVIEW

File: exercises/practice/rail-fence-cipher/.meta/example.lisp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+(defstruct post letter row column)
+
+(defun make-fence (seq-length rails)
+  "Function make-fence will output a list of structs of type post, which will come
+to represent a character and its coordinates in the fence.  Function contains a
+loop that goes along the columns, and uses mathematical equations to find the
+correct row.  Letters are not filled in here, and are set to nil."
+  (loop for i from 0 below seq-length with r = (1- rails)
+    if (< (mod i (* 2 r)) r) ; if i mod 2r < r then...
```
Comment (verdammelt) [inline exercises/practice/rail-fence-cipher/.meta/example.lisp:25]: I suggest removing these comments - I don't think they are actually helping are they?

File: exercises/practice/rail-fence-cipher/.meta/example.lisp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    collect (make-post :letter nil :column i :row (mod i r)) ; row = i mod r, else...
+    else
+    collect (make-post :letter nil :column i :row (- r (mod i r))))) ; row = r - (i mod r)
+
+(defun populate-and-sort (msg fence sort-key)
```
Comment (verdammelt) [inline exercises/practice/rail-fence-cipher/.meta/example.lisp:30]: Feels like this function is doing too much, it populates the fence, sorts the fence and then unpacks the fence.

Maybe if it was going to do all of that maybe it should just make the fence as well? (yes i know that won't work in the way you wrote decode).

I wonder if maybe populate-and-sort should do less? Maybe unpacking a fence is the responsibility of some other function? Perhaps we need 4 functions in the fence interface: a constructor, a writer (string->fence), a sorter (fence -> fence), and a reader (fence -> string).

File: exercises/practice/rail-fence-cipher/.meta/example.lisp (hunk 1 above)
Comment (PaulT89) [reply exercises/practice/rail-fence-cipher/.meta/example.lisp:25]: For me they make it clear at a glance - but that's just me.  I have trouble with prefix notation arithmetic.

File: exercises/practice/rail-fence-cipher/.meta/example.lisp (hunk 2 above)
Comment (PaulT89) [reply exercises/practice/rail-fence-cipher/.meta/example.lisp:30]: The reason it does so much is mostly just to decrease duplication, but I can split it into 3 separate functions if you wish.

**Review by verdammelt — APPROVED** [review]

**Pull request merged**